src/Data/Form.purs

```diff
@@ -176,50 +254,91 @@ eval get page = do
       placeholder <- evalExpr get input.placeholder
       required <- evalExpr get input.required
       value <- traverse (evalExpr get) input.value
-      pure
-        ( Currency
-          { default
-          , placeholder
-          , required
-          , value
-          }
-        )
+      pure $ validate $ Currency
+        { default
+        , placeholder
+        , required
+        , value
+        , errors: mempty
+        }
     Dropdown input -> do
       default <- traverse (evalExpr get) input.default
       options <- evalExpr get input.options
       placeholder <- evalExpr get input.placeholder
       required <- evalExpr get input.required
-      value <- traverse (evalExpr get) input.value
-      pure
-        ( Dropdown
-          { default
-          , options
-          , placeholder
-          , required
-          , value
-          }
-        )
+      value' <- traverse (evalExpr get) input.value
+      let arrayOptions = Data.Maybe.fromMaybe [] (toArray options)
+          value = case value' of
+            UserInput x ->
+              if (not $ x `Data.Array.elem` arrayOptions)
```

Comment:
@joneshf-cn I'm not 💯 sold on fully namespacing imported functions like this. I do agree with people on Slack who feel like it adds visual noise, though I also agree it can be useful for clarity. But I also feel `Array.elem` would be just as useful here. It also seems hard to enforce, or consistently do. But since I've seen you use it here and there in this project, I figured I'd test-drive it, see if I don't come around 🙂

Reply:
Heh, no worries. I've tried not to do it too much, because that's not been the style here and I don't feel like I should be changing the style around a bunch. Don't feel pressured to do it just because I'm working on this package. I'm more than happy to follow whatever style we've got.